test/functional/feature_framework_startup_failures.py

```diff
@@ -0,0 +1,109 @@
+We should maintain this bar of only raising one exception as long as
+additional maintenance and complexity is low.
+
+Test relaunches itself into a child processes in order to trigger failure
+without the parent process' BitcoinTestFramework also failing.
+"""
+
+from test_framework.util import rpc_port
+from test_framework.test_framework import BitcoinTestFramework
+
+import re
+import subprocess
+import sys
+
+class FeatureFrameworkRPCFailure(BitcoinTestFramework):
+    def set_test_params(self):
+        # Only run a node for child processes
+        self.num_nodes = 1 if any(o is not None for o in [self.options.internal_rpc_timeout,
+                                                          self.options.internal_extra_args,
+                                                          self.options.internal_start_stop]) else 0
+
+        if self.options.internal_rpc_timeout is not None:
+            self.rpc_timeout = self.options.internal_rpc_timeout
+        if self.options.internal_extra_args:
+            self.extra_args = [[self.options.internal_extra_args]]
+
+    def add_options(self, parser):
+        parser.add_argument("--internal-rpc_timeout", dest="internal_rpc_timeout", help="ONLY TO BE USED WHEN TEST RELAUNCHES ITSELF")
+        parser.add_argument("--internal-extra_args", dest="internal_extra_args", help="ONLY TO BE USED WHEN TEST RELAUNCHES ITSELF")
+        parser.add_argument("--internal-start_stop", dest="internal_start_stop", help="ONLY TO BE USED WHEN TEST RELAUNCHES ITSELF")
+
+    def setup_network(self):
+        # Avoid doing anything if num_nodes == 0, otherwise we fail.
+        if self.num_nodes > 0:
+            if self.options.internal_start_stop:
+                self.add_nodes(self.num_nodes, self.extra_args)
+                self.nodes[0].start()
+                self.nodes[0].stop_node()
+            else:
+                BitcoinTestFramework.setup_network(self)
+
+    def _run_test_internal(self, args, expected_exception):
+        try:
+            result = subprocess.run([sys.executable, __file__] + args, encoding="utf-8", stdout=subprocess.PIPE, stderr=subprocess.STDOUT, timeout=60 * self.options.timeout_factor)
+        except subprocess.TimeoutExpired as e:
+            print(f"Unexpected timeout, subprocess output:\n{e.output}\nSubprocess output end", file=sys.stderr)
+            raise
+
+        success = True
+
+        traceback_count = len(re.findall("Traceback", result.stdout))
+        if traceback_count != 1:
+            self.log.error(f"Found {traceback_count}/1 tracebacks - expecting exactly one with no knock-on exceptions.")
+            success = False
+
+        matching_exception_count = len(re.findall(expected_exception, result.stdout))
+        if matching_exception_count != 1:
+            self.log.error(f"Found {matching_exception_count}/1 occurrences of the specific exception: {expected_exception}")
+            success = False
```

Comment:
we could likely use the walrus operator here to narrow the scope:
```suggestion
		if (matches := len(re.findall(expected_exception, result.stdout))) != 1:
		    self.log.error(f"Found {matches}/1 occurrences of the specific exception: {expected_exception}")
		    success = False
```

Reply:
Prefer using the walrus for truthiness tests which don't require the additional `() != 1`.

Reply:
The point is to narrow the scope of variables so that we can give them more focused names.
The example in the doc https://docs.python.org/3/whatsnew/3.8.html is basically this exact escenario
```python
if (n := len(a)) > 10:
    print(f"List is too long ({n} elements, expected <= 10)")
```

Reply:
Or we could avoid repetition as well by a more functional approach of separating data from algorithm:
```python
patterns = {
    "Traceback": "tracebacks - expecting exactly one with no knock-on exceptions",
    expected_exception: f"occurrences of the specific exception: {expected_exception}",
    "Test failed. Test logging available at": "test failure output messages"
}
errors = [f"Found {count}/1 {message}."
          for pattern, message in patterns.items()
          if (count := len(re.findall(pattern, output))) != 1]
```